pkg/console/controllers/route/controller.go

```diff
@@ -174,7 +174,16 @@ func (c *RouteSyncController) removeRoute(routeName string) error {
 }
 
 func (c *RouteSyncController) SyncDefaultRoute(operatorConfig *operatorsv1.Console) (*routev1.Route, string, error) {
-	requiredDefaultRoute := routesub.DefaultRoute(operatorConfig)
+	customTLSSecret, configErr := c.ValidateDefaultRouteConfig(operatorConfig)
+	if configErr != nil {
+		return nil, "InvalidDefaultRouteConfig", configErr
+	}
+	customTLSCert, secretValidationErr := ValidateCustomCertSecret(customTLSSecret)
+	if secretValidationErr != nil {
+		return nil, "InvalidCustomTLSSecret", secretValidationErr
+	}
+
+	requiredDefaultRoute := routesub.DefaultRoute(operatorConfig, customTLSCert)
```

Comment:
> Is this unconditionally applying the custom certificate to the default route?

No the custom cert will be applied to the default route only if the `spec.route.secret.name` is set (`spec.route.hostname` shall not be set in this case) and pointing to a secret in `openshift-config` namespace.

When cluster-admin sets the `spec.route.hostname` with a duplicate of the default hostname, nothing should happen, since a custom route with dup. hostname wont be admitted.